src/graphics/program-lib/chunks/common/frag/gamma2_2.js

```diff
@@ -1,40 +1,21 @@
 export default /* glsl */`
-vec3 gammaCorrectInput(vec3 color) {
-    return pow(color, vec3(2.2));
-}
-
 float gammaCorrectInput(float color) {
-    return pow(color, 2.2);
-}
-
-vec4 gammaCorrectInput(vec4 color) {
-    return vec4(pow(color.rgb, vec3(2.2)), color.a);
-}
-
-vec4 texture2DSRGB(sampler2D tex, vec2 uv) {
-    vec4 rgba = texture2D(tex, uv);
-    rgba.rgb = gammaCorrectInput(rgba.rgb);
-    return rgba;
+    return decodeGamma(color);
 }
 
-vec4 texture2DSRGB(sampler2D tex, vec2 uv, float bias) {
-    vec4 rgba = texture2D(tex, uv, bias);
-    rgba.rgb = gammaCorrectInput(rgba.rgb);
-    return rgba;
+vec3 gammaCorrectInput(vec3 color) {
+    return decodeGamma(color);
 }
 
-vec4 textureCubeSRGB(samplerCube tex, vec3 uvw) {
-    vec4 rgba = textureCube(tex, uvw);
-    rgba.rgb = gammaCorrectInput(rgba.rgb);
-    return rgba;
+vec4 gammaCorrectInput(vec4 color) {
+    return vec4(decodeGamma(color.xyz), color.w);
 }
 
 vec3 gammaCorrectOutput(vec3 color) {
-    #ifdef HDR
+#ifdef HDR
     return color;
-    #else
-    color += vec3(0.0000001);
-    return pow(color, vec3(0.45));
-    #endif
+#else
+    return pow(color + 0.0000001, vec3(1.0 / 2.2));
```

Comment:
doesn't need to be vec3(0.0000001) ?

Reply:
nope this works. the float is added to vector components.